arangod/Aql/OptimizerRules.cpp

```diff
@@ -1439,7 +1439,7 @@ class PropagateConstantAttributesHelper {
                   // don't remove a smart join attribute access!
                   return;
                 } else {
-                  std::vector<std::string> const& shardKeys = logical->shardKeys();
+                  std::vector<std::string> const& shardKeys = collection->shardKeys(true);
```

Comment:
```suggestion
                  std::vector<std::string> shardKeys = collection->shardKeys(true);
```
`LogicalCollection::shardKeys()` returns a `std::vector<std::string> const&`, but `Collection::shardKeys()` returns a `std::vector<std::string>`.